src/briefcase/commands/dev.py

```diff
@@ -73,6 +73,13 @@ def add_options(self, parser):
             action="store_true",
             help="Run the app in test mode",
         )
+        parser.add_argument(
+            "--no-isolation",
+            dest="no_isolation",
+            action="store_true",
+            default=False,
+            help="Run without creating an isolated environment",
+        )
```

Comment:
We need to be careful about exposing arguments that can't be used. At least for now, this argument should only be defined on the *web* dev instance.